core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -186,8 +185,8 @@ object Uri extends UriFunctions {
 
 
   private def renderParamsAndFragment(writer: Writer, p: Query, f: Option[Fragment]): writer.type = {
-    if (p.nonEmpty) writer << '?' << p
-    if (f.isDefined) writer << '#' << UrlCodingUtils.urlEncode(f.get, spaceIsPlus = false)
+    if (!p.isEmpty) writer << p
```

Comment:
Ya, I think  I have a preference for `Option[Query]` if that's possible, if not I would be curious why it's not possible.